### PR TITLE
DSE-29039 Dynamic Population of engine->runtime mapping

### PR DESCRIPTION
**Pull request by pratapdesai**
This change helps in populating the Legacy engine to Target Runtime mapping on demand. Meaning, the runtime versions existing in the  Target Workspace will be automatically picked and used by the mapping.

### CODE REVIEW

File: cmlutils/project_entrypoint.py [hunk 1]
```diff
@@ -245,3 +246,60 @@ def project_import_cmd(project_name):
...
+            break
+        runtimes = runtimes + response.get("runtimes", [])
+        page_token = response.get("next_page_token", "")
+
+    if len(runtimes) > 0:
+        legacy_runtime_image_map = parse_runtimes_v2(runtimes)
+    else:
+        return
+
+    with open('legacy_engine_runtime_constants.py', 'w') as legacy_engine_runtime_constants:
```
Comment (pratapdesai) [inline cmlutils/project_entrypoint.py:304]: Test it using branch and identify where the utility is stored.

File: README.md [hunk 2]
```diff
@@ -134,10 +134,19 @@ The legacy engine is transitioned to an ML runtime based on the table below.
 | r | ml-runtime-workbench-r4.1-standard:2023.05.2-b7 |
 | scala | ml-runtime-workbench-scala2.11-standard:2023.05.2-b7 |
 
-The mapping can be found in the `cmlutils/constants.py` file, and if you want to make changes to it, you can either modify the constants.py file directly or modify the `runtime_identifier` entry in the metadata files generated after using the export command.
+The mapping can be found in the `cmlutils/legacy_engine_runtime_constants.py` file, and if you want to make changes to it, you can either modify the constants.py file directly or modify the `runtime_identifier` entry in the metadata files generated after using the export command.
```
Comment (navinag1989) [inline README.md:137]: small correction "constants.py" to  "legacy_engine_runtime_constants.py"

File: README.md [hunk 3]
```diff
@@ -134,10 +134,19 @@ The legacy engine is transitioned to an ML runtime based on the table below.
...
+
+ #### Dynamic Population of Legacy engine runtime mapping
+
+The default mapping existing in the `cmlutils/legacy_engine_runtime_constants.py` can be dynamically populated based on the runtimes available in the target (CML) workload. Please use this command on your linux/mac machine:
```
Comment (navinag1989) [inline README.md:141]: Add prerequisite of populating import-config.ini file.

File: cmlutils/project_entrypoint.py [hunk 4]
```diff
@@ -245,3 +246,60 @@ def project_import_cmd(project_name):
...
+
+    username = config[USERNAME_KEY]
+    url = config[URL_KEY]
+    apiv1_key = config[API_V1_KEY]
+    local_directory = config[OUTPUT_DIR_KEY]
+    ca_path = config[CA_PATH_KEY]
+
+    local_directory = get_absolute_path(local_directory)
+    ca_path = get_absolute_path(ca_path)
+
```
Comment (navinag1989) [inline cmlutils/project_entrypoint.py:273]: Can you configure logging here and logs at various stages? 
_configure_project_command_logging(log_filedir, project_name)

**Review by navinag1989 — APPROVED** [review]

**Pull request merged**